oak-core/src/test/java/org/apache/jackrabbit/oak/query/QueryInformationLoggerTest.java

```diff
@@ -0,0 +1,143 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.slf4j.event.Level;
+
+public class QueryInformationLoggerTest {
+    
+    private static final Logger LOG = LoggerFactory.getLogger(QueryInformationLoggerTest.class);
+
+    private static final String EXTERNAL_LOG = QueryInformationLogger.class.getName() + ".externalQuery";
+    private static final String INTERNAL_LOG = QueryInformationLogger.class.getName() + ".internalQuery";
+    
+    String[] ignoredClasses = new String[] {"org.apache.jackrabbit.oak","java.lang","sun.reflect", "jdk"};
+    String[] allClassesIgnored = new String[] {"java","org","net","sun","jdk"};
+    
+    
+    @Test
+    public void regularQueryTest() {
+        
+        LogCustomizer external = LogCustomizer.forLogger(EXTERNAL_LOG).enable(Level.INFO).create();
+        LogCustomizer internal = LogCustomizer.forLogger(INTERNAL_LOG).enable(Level.TRACE).create();
+        try {
+            external.starting();
+            internal.starting();
+            String query = "SELECT * FROM [cq:Page]";
+            QueryInformationLogger.logCaller(query, ignoredClasses);
+            assertEquals("Exactly 1 entry must be written",1,external.getLogs().size());
+            String logEntry = external.getLogs().get(0);
+            assertTrue(logEntry.contains("org.junit.runners.model")); // org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
+            assertEquals(0,internal.getLogs().size());
+        } finally {
+            external.finished();
+            internal.finished();
+        }      
+    }
+    
+    @Test
+    public void withStacktrace() {
+        
+        LogCustomizer external = LogCustomizer.forLogger(EXTERNAL_LOG).enable(Level.DEBUG).create();
+        LogCustomizer internal = LogCustomizer.forLogger(INTERNAL_LOG).enable(Level.TRACE).create();
+        try {
+            external.starting();
+            internal.starting();
+            String query = "SELECT * FROM [cq:Page]";
+            QueryInformationLogger.logCaller(query, ignoredClasses);
+            assertEquals("Exactly 2 entries must be written",2,external.getLogs().size()); //     
+            assertEquals(0,internal.getLogs().size());
+        } finally {
+            external.finished();
+            internal.finished();
+        }      
+    }
+    
+    @Test
+    public void testOakInternalStatement() {
+        LogCustomizer external = LogCustomizer.forLogger(EXTERNAL_LOG).enable(Level.DEBUG).create();
+        LogCustomizer internal = LogCustomizer.forLogger(INTERNAL_LOG).enable(Level.INFO).create();
+        try {
+            external.starting();
+            internal.starting();
+            String query = "SELECT * FROM [cq:Page] " + QueryInformationLogger.OAK_INTERNAL_MARKER;
+            QueryInformationLogger.logCaller(query, ignoredClasses);
+            
+            // On INFO no logs are written
+            assertEquals(0,external.getLogs().size()); //     
```

Comment:
- formatting: add a space after ,
- i think the // at the end of the line are not needed, right? the comment is already above